clients/geth/specular/rollup/services/base.go

```diff
@@ -33,12 +35,31 @@ type BaseService struct {
 	Wg     sync.WaitGroup
 }
 
+func connectL1(ctx context.Context, endpoint string) (*ethclient.Client, error) {
+	var l1 *ethclient.Client
+	var err error
+	retryOpts := []retry.Option{
+		retry.Context(ctx),
+		retry.Attempts(3),
```

Comment:
we could also consider just letting it retry perpetually every x seconds until it connects

Reply:
its not friendly to ci/cd if l1 fails

Reply:
yeah actually what you have makes more sense